### ISSUE TITLE
pentesting and bugfixing

### ISSUE THREAD

**Issue by gnosticDeveloper** [gnosticDeveloper/kenoma#170]
a new pentesting pass should be done and any outstanding issues are to be solved. issues with database grants from openbao are likely, since they have not been properly tested for accessibility yet
